### PR TITLE
Create people migration

### PR DESCRIPTION
**Pull request by SimonLab**
- Create people table with `person.id` and `name`
- Copy existing `person_id` from `items` and `tags`
- Add foreign key using references in `items` and `tags`

### CODE REVIEW

**Comment by SimonLab** [comment]
This PR add a simple profile page.
I have created a new migration to create the `people` table and the `person` schema linked to this table.
To keep the existing data, I run an sql command which create new person based on the existing person_id from items and tags.

Because we are going to use the unique name of the logged in person, we are redirecting the user to the edit profile page if the name is not defined yet.

File: lib/app_web/templates/layout/root.html.heex [hunk 1]
```diff
@@ -16,7 +16,9 @@
       <div class="container flex flex-wrap justify-between items-center mx-auto">
 
         <%= if @loggedin do %>
+          <a href={Routes.profile_path(@conn, :edit, @person.id)}>
```
Comment (SimonLab) [inline lib/app_web/templates/layout/root.html.heex:19]: Add link to the profile picture in the nav bar to the profile edit page

File: priv/repo/migrations/20221004195054_add_person.exs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    create table(:people, primary_key: false) do
+      add(:person_id, :integer, primary_key: true)
+      add(:name, :citext)
+
+      timestamps()
+    end
+
+    # insert new people rows by select unique person_id from items and tags
+    execute(
```
Comment (SimonLab) [inline priv/repo/migrations/20221004195054_add_person.exs:15]: Create person from the existing person_id from the items and tags table

File: lib/app/person.ex [hunk 3]
```diff
@@ -0,0 +1,39 @@
+defmodule App.Person do
+  use Ecto.Schema
+  import Ecto.Changeset
+  alias App.{Item, Repo, Tag}
+  alias __MODULE__
+  # https://hexdocs.pm/phoenix/Phoenix.Param.html
```
Comment (nelsonic) [inline lib/app/person.ex:6]: I find the docs for `Param` pretty useless. Wish there was a standalone example. 💭

File: lib/app/person.ex [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    has_many :tags, Tag, foreign_key: :person_id
+    timestamps()
+  end
+
+  @doc false
+  def changeset(person, attrs \\ %{}) do
+    person
+    |> cast(attrs, [:person_id, :name])
+    |> validate_required([:person_id, :name])
+    |> unique_constraint(:name, name: :people_name_index)
```
Comment (nelsonic) [inline lib/app/person.ex:23]: Can't say I understand why the `:name` should be unique ... the `:person_id`, sure, but name ...? 🤷‍♂️

File: lib/app_web/controllers/profile_controller.ex [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  def update(
+        conn,
+        %{"person_id" => person_id, "person" => person_params}
+      ) do
+    profile = Person.get_person!(person_id)
+
+    case Person.update_person(profile, person_params) do
+      {:ok, _person} ->
+        conn
+        |> put_flash(:info, "Person updated successfully.")
```
Comment (nelsonic) [inline lib/app_web/controllers/profile_controller.ex:21]: Guessing this "Person updated successfully." is auto-generated? 🤖 
Maybe we could change it to "Your profile was updated." or something a bit more friendly? 💭

File: lib/app/person.ex (hunk 4 above)
Comment (SimonLab) [reply lib/app/person.ex:23]: I'm thinking of using the name later on in the UI to select people to add to groups. If the name is not unique we won't be able to distinguish between users. Maybe it should be called `username`.

File: lib/app_web/controllers/profile_controller.ex (hunk 5 above)
Comment (SimonLab) [reply lib/app_web/controllers/profile_controller.ex:21]: The flash messages are not displayed at the moment.
The function to display them might have been removed previously, I need to have a look at it but we can definitely update the message

**Review by nelsonic — APPROVED** [review]
@SimonLab looks good. 👌 
Only one note on the uniqueness of `person.name` 💭 
But fairly certain we can modify this later when needed.
my first and last name are not unique by any stretch ... 🙄 

image

**Pull request merged**